File: LicenseManagementApi/Models/Entities/ApiKey.cs
namespace LicenseManagementApi.Models.Entities;

public class ApiKey
{
    public Guid Id { get; set; }
    public string KeyHash { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? LastUsedAt { get; set; }
}

File: vue-ui-frontend/src/types/apiKey.ts
export interface ApiKey {
  id: string
  name: string
  role: string
  isActive: boolean
  createdBy: string
  createdAt: string
  lastUsedAt: string | null
}

export interface CreateApiKeyDto {
  name: string
  role: string
  createdBy: string
}

export interface UpdateApiKeyDto {
  name: string
  role: string
  isActive: boolean
}

export interface ApiKeyCreationResponse {
  id: string
  name: string
  role: string
  isActive: boolean
  createdBy: string
  createdAt: string
  key: string // Only available on creation
}
